src/Tests/Microsoft.NET.Sdk.BlazorWebAssembly.Tests/WasmJsModulesIntegrationTests.cs

```diff
@@ -0,0 +1,342 @@
+            });
+
+
+            var publish = new PublishCommand(ProjectDirectory);
+            publish.WithWorkingDirectory(ProjectDirectory.TestRoot);
+            var publishResult = publish.Execute("/bl");
+            publishResult.Should().Pass();
+
+            var outputPath = publish.GetOutputDirectory(DefaultTfm).ToString();
+            var intermediateOutputPath = publish.GetIntermediateDirectory(DefaultTfm, "Debug").ToString();
+
+            var path = Path.Combine(intermediateOutputPath, "StaticWebAssets.publish.json");
+            new FileInfo(path).Should().Exist();
+            var manifest = StaticWebAssetsManifest.FromJsonBytes(File.ReadAllBytes(path));
+            AssertManifest(manifest, LoadPublishManifest());
+
+            var buildLibrary = GetLibraryInitializer(Path.Combine(intermediateOutputPath, "blazor.boot.json"));
+            var publishLibrary = GetLibraryInitializer(Path.Combine(intermediateOutputPath, "blazor.publish.boot.json"));
+
+            publishLibrary.GetString().Should().NotBe(buildLibrary.GetString());
+
+            new FileInfo(Path.Combine(outputPath, "wwwroot", "blazorwasm-minimal.modules.json")).Should().NotExist();
+            var lib = new FileInfo(Path.Combine(outputPath, "wwwroot", "blazorwasm-minimal.lib.module.js"));
+            lib.Should().Exist();
+
+            var wwwrootPublishLibrary = GetLibraryInitializer(Path.Combine(outputPath, "wwwroot", "_framework", "blazor.boot.json"));
+            publishLibrary.GetString().Should().Be(wwwrootPublishLibrary.GetString());
+
+            AssertPublishAssets(
+                StaticWebAssetsManifest.FromJsonBytes(File.ReadAllBytes(path)),
+                outputPath,
+                intermediateOutputPath);
+
+            static JsonElement GetLibraryInitializer(string path)
+            {
+                var blazorBootJson = new FileInfo(path);
+                blazorBootJson.Should().Exist();
+                var contents = JsonSerializer.Deserialize<JsonDocument>(blazorBootJson.OpenRead());
+                contents.RootElement.TryGetProperty("resources", out var resources).Should().BeTrue();
+                resources.TryGetProperty("libraryInitializers", out var initializers).Should().BeTrue();
+                initializers.TryGetProperty("blazorwasm-minimal.lib.module.js", out var buildLibrary).Should().BeTrue();
+                return buildLibrary;
+            }
+        }
+
+        [Fact]
+        public void JsModules_CanCustomizeBlazorInitialization()
+        {
+            // Arrange
+            var testAsset = "BlazorWasmMinimal";
+            ProjectDirectory = CreateAspNetSdkTestAsset(testAsset);
+            File.WriteAllText(Path.Combine(ProjectDirectory.TestRoot, "wwwroot", "blazorwasm-minimal.lib.module.js"), "console.log('Hello initializer')");
+
+            ProjectDirectory.WithProjectChanges(document =>
+            {
+                document.Root.Add(new XElement("Target",
```

Comment:
I think you're better off using `XElement.Parse` at this point. It's a bit difficult to follow what's happening here.

Reply:
done